info.puredata.Pd.yml

```diff
@@ -0,0 +1,74 @@
+app-id: info.puredata.Pd
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: pd-gui
+finish-args:
+# Display
+  - --share=ipc
+  - --socket=x11
+  - --device=dri
+# Audio
+  - --socket=pulseaudio
+  - --share=network
+  - --filesystem=xdg-run/pipewire-0
+# Filesystem access
+  - --filesystem=home
+# MIDI, webcams, controllers,...
+  - --device=all
+modules:
+  - shared-modules/linux-audio/jack2.json
+  - name: tcl
+    buildsystem: autotools
+    subdir: unix
+    build-options:
+      no-debuginfo: true
+    sources:
+      - type: archive
+        url: https://prdownloads.sourceforge.net/tcl/tcl8.6.13-src.tar.gz
+        sha256: 43a1fae7412f61ff11de2cfd05d28cfc3a73762f354a417c62370a54e2caf066
+        x-checker-data:
+          type: anitya
+          project-id: 4941
+          stable-only: true
+          url-template: https://prdownloads.sourceforge.net/tcl/tcl$version-src.tar.gz
+  - name: tk
+    buildsystem: autotools
+    subdir: unix
+    build-options:
+      no-debuginfo: true
+    post-install:
+      - test -e /app/bin/wish || find /app/bin/ -name "wish*.*" -printf "%f\0/app/bin/wish" -quit | xargs -0 ln -s
+    sources:
+      - type: archive
+        url: https://prdownloads.sourceforge.net/tcl/tk8.6.13-src.tar.gz
+        sha256: 2e65fa069a23365440a3c56c556b8673b5e32a283800d8d9b257e3f584ce0675
+        x-checker-data:
+          type: anitya
+          project-id: 11426
+          stable-only: true
+          url-template: https://prdownloads.sourceforge.net/tcl/tk$version-src.tar.gz
+
+  - name: puredata
+    buildsystem: autotools
+    config-opts:
+      - --enable-jack
+    sources:
+      - type: git
+        url: https://github.com/pure-data/pure-data.git
+        tag: 0.54-0
+        commit: c0a7c16c63e86ccc8ce05c8273ccd8aec7ae044c
+        x-checker-data:
+          type: git
+          tag-pattern: "^([\\d.]+-[\\d]+)$"
+          is-main-source: true
+  - name: integration
+    buildsystem: simple
+    build-commands:
+      - mv "/app/share/icons/hicolor/512x512/apps/puredata.png"    "/app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png"
+      - mv "/app/share/icons/hicolor/48x48/apps/puredata.png"      "/app/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png"
+      - mv "/app/share/icons/hicolor/scalable/apps/puredata.svg"   "/app/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg"
+      - mv "/app/share/applications/org.puredata.pd-gui.desktop"   "/app/share/applications/${FLATPAK_ID}.desktop"
+      - sed -e "s|^Icon=.*|Icon=${FLATPAK_ID}|"                 -i "/app/share/applications/${FLATPAK_ID}.desktop"
+      - mv "/app/share/metainfo/org.puredata.pd-gui.metainfo.xml"  "/app/share/metainfo/${FLATPAK_ID}.metainfo.xml"
+      - sed -e "s|org\.puredata\.pd-gui|${FLATPAK_ID}|g"        -i "/app/share/metainfo/${FLATPAK_ID}.metainfo.xml"
```

Comment:
actually changing the id which this also does should be done with `appstream-util`
```
appstream-util modify $FLATPAK_DEST/share/metainfo/${FLATPAK_ID}.metainfo.xml id ${FLATPAK_ID}
```

Reply:
i tend to disagree.
internally the application consists of two components: Pd-core and Pd-gui, which are separate programs that communicate via a network connection.
the current metainfo.xml comes from the Debian packaging, where the gui and the core are available in separate packages, and the metainfo.xml is shipped alongside the gui - hence the name.

for flatpak, the two components are shipped in a single flatpak application, which (imho) should be made available under a name that people will understand.
this could be either
- `info.puredata.Pd`
- `info.puredata.PureData`

Reply:
no. whatever is used upstream should be used here. Unless it's invalid (it's not).

Reply:
so i think i'm going to change this upstream.

am i right in assuming that changing the app-id will basically mean a new application, both from the user perspective (as in: there will be two distinct applications (with similar descriptions) available on flathub, in software centers,...) and from a dev perspective (i would need to re-issue an add request and a separate repository will be created)?

so if we plan to change the app id, we should do it *before* entering flathub.

Reply:
otoh, i'm sure we won't be able to get a release very soon (but of course: you never know...)

I'm about to push a  commit to our (that is: upstream's) `develop` branch. would that be enough to accept local (that is: flatpak) workarounds that preempt the new upstream ID?

or do we have to stall this until there's a new upstream release with the fixed IDs?